Runtime/Tracking/Collision/CollisionTracker.cs

```diff
@@ -64,7 +113,87 @@ protected virtual void OnTriggerExit(Collider collider)
                 return;
             }
 
+            RemoveTrackedNode(collider);
             OnCollisionStopped(eventData.Set(this, true, null, collider));
         }
+
+        /// <summary>
+        /// Adds a <see cref="CollisionTrackerDisabledObserver"/> to the <see cref="GameObject"/> that contains the <see cref="Collider"/> causing the collision.
+        /// </summary>
+        /// <param name="target">The target to add the <see cref="CollisionTrackerDisabledObserver"/> component to.</param>
+        protected virtual void AddTrackedNode(Collider target)
+        {
+            if (target == null || !StopCollisionsOnDisable)
+            {
+                return;
+            }
+
+            trackedCollisions.Add(target);
+            CollisionTrackerDisabledObserver node = target.gameObject.AddComponent<CollisionTrackerDisabledObserver>();
+            node.Source = this;
+            node.Target = target;
+        }
+
+        /// <summary>
+        /// Removes the <see cref="CollisionTrackerDisabledObserver"/> from the <see cref="GameObject"/> that contains the <see cref="Collider"/> ceasing the collision.
+        /// </summary>
+        /// <param name="target">The target to remove the <see cref="CollisionTrackerDisabledObserver"/> component from.</param>
+        protected virtual void RemoveTrackedNode(Collider target)
+        {
+            if (target == null || !StopCollisionsOnDisable)
+            {
+                return;
+            }
+
+            foreach (CollisionTrackerDisabledObserver node in target.gameObject.GetComponents<CollisionTrackerDisabledObserver>().ToArray())

```

Comment:
```suggestion
            foreach (CollisionTrackerDisabledObserver observer in target.gameObject.GetComponents<CollisionTrackerDisabledObserver>())
```
Not sure why that `.ToArray` was used, but it seems unnecessary.

Reply:
`ToArray` was just a copy/paste from another foreach. I've deleted it as it isn't needed.